reactor-test/src/main/java/reactor/test/scheduler/VirtualTimeScheduler.java

```diff
@@ -399,8 +400,12 @@ final void drain() {
 					queue.poll();
 
 					// Only execute if not unsubscribed
-					if (!current.scheduler.shutdown) {
-						current.run.run();
+					if (!current.worker.shutdown) {
+                        try {
+                            current.run.run();
+                        } finally {
+                            current.set(true);
```

Comment:
I'm wondering about that one given that the documentation of `schedule` methods explains what the `Disposable` is supposed to mean: it can be used to cancel the task and `isDisposed()` in case the task is cancelled.

At the very least, since it is inconsistent with the behavior of prod schedulers, this would need to be documented. the `schedulePeriodically` method should have a similar documentation addendum that states that `isDisposed()` can never mean the task is done, because the only way for a periodic task to be done is to cancel it (otherwise it continuously reschedule itself).

Reply:
good point. `SchedulerTask`, used eg. in `ParallelScheduler`, does that as well.

for periodicTask, I was pushing for a documentation of the fact it behaves slightly differently from the other `Disposable`-returning methods, in the sense that the only thing that can mark a periodic task as `isDisposed()` is to cancel it. but it should be ok as it is.